pennylane_lightning/lightning_tensor/_tensornet.py

```diff
@@ -31,44 +31,100 @@
     Ms = [[] for _ in range(n_wires)]
+    site_len = np.prod(site_shape)
+    psi = np.reshape(psi, (site_len, -1))  # split psi [2, 2, 2, 2...] to psi [site_len, -1]
 
-    psi = np.reshape(psi, (2, -1))  # split psi[2, 2, 2, 2..] = psi[2, (2x2x2...)]
-    U, Vd = svd_split(psi, bond_dim)  # psi[2, (2x2x..)] = U[2, mu] Vd[mu, (2x2x2x..)]
+    U, Vd = svd_split(
+        psi, site_shape, max_bond_dim
+    )  # psi [site_len, -1] -> U [site_len, mu] Vd [mu, (2x2x2x..)]
 
-    Ms[0] = U
+    Ms[0] = U.reshape(tuple(site_shape + [-1]))
     bondL = Vd.shape[0]
     psi = Vd
 
     for i in range(1, n_wires - 1):
-        psi = np.reshape(psi, (2 * bondL, -1))  # reshape psi[2 * bondL, (2x2x2...)]
-        U, Vd = svd_split(psi, bond_dim)  # psi[2, (2x2x..)] = U[2, mu] Vd[mu, (2x2x2x..)]
+        psi = np.reshape(psi, (site_len * bondL, -1))  # reshape psi[site_len*bondL, -1]
+        U, Vd = svd_split(
+            psi, site_shape, max_bond_dim
+        )  # psi [site_len*bondL, -1] -> U [site_len, mu] Vd [mu, (2x2x2x..)]
         Ms[i] = U
 
         psi = Vd
         bondL = Vd.shape[0]
 
-    Ms[n_wires - 1] = Vd
+    Ms[-1] = Vd.reshape(tuple([-1] + site_shape))
 
     return Ms
 
 
+def gate_matrix_decompose(gate_ops_matrix, wires, c_dtype):
+    """Permute and decompose a gate matrix into MPO sites. This method return the MPO sites in the Fortran order of the ``cutensornet`` backend. Note that MSB in the Pennylane convention is the LSB in the ``cutensornet`` convention."""
+    sorted_indexed_wires = sorted(enumerate(wires), key=lambda x: x[1])
+
+    sorted_wires = []
+    original_axes = []
+    for index, wire in sorted_indexed_wires:
+        sorted_wires.append(wire)
+        original_axes.append(index)
+
+    tensor_shape = [2] * len(wires) * 2
+
+    matrix = gate_ops_matrix.astype(c_dtype)
+
+    # Convert the gate matrix to the correct shape and complex dtype
+    gate_tensor = matrix.reshape(tensor_shape)
+
+    # Create the correct order of indices for the gate tensor to be decomposed
+    indices_order = []
+    for i in range(len(wires)):
+        indices_order.extend([original_axes[i], original_axes[i] + len(wires)])
+    # Reverse the indices order to match the target wire order of cutensornet backend
+    indices_order.reverse()
+
+    # Permutation of the gate tensor
+    gate_tensor = np.transpose(gate_tensor, axes=indices_order)
+
+    mpo_site_shape = [2] * 2
+    # TODO: Discuss if public interface for max_mpo_bond_dim argument
+    max_mpo_bond_dim = 2 ** len(wires)  # Exact SVD decomposition for MPO
+    # The indices order of MPOs: 1. left-most site: [ket, bra, bondR]; 2. right-most sites: [bondL, ket, bra]; 3. sites in-between: [bondL, ket, bra, bondR].
+    MPOs = decompose_dense(gate_tensor, len(wires), mpo_site_shape, max_mpo_bond_dim)
+
+    # Convert the MPOs to the correct order for the cutensornet backend
+    mpos = []
+    for i in range(len(MPOs)):
```

Comment:
I thought @AmintorDusko suggested to use `enumerator`... I cannot find his comment anymore; any blocker with his suggestion?

Reply:
Suggestions accepted! I guess you were on an old commit